### PR TITLE
nrf_modem_lib: removal of deprecated socket dfu

### PR DESCRIPTION
**Pull request by anhmolt**
Remove deprecated API translations for socket dfu.

Signed-off-by: Andreas Moltumyr <andreas.moltumyr@nordicsemi.no>

_Commit: 9121b9c2e6272297e16c79a61c4b73a5ee1f3eb5 to be removed after [nrfconnect/sdk-nrf#6055](https://github.com/nrfconnect/sdk-nrf/pull/6055) is merged and included into lemrey/sdk-nrf_

### CODE REVIEW

File: subsys/dfu/dfu_target/src/dfu_target_modem_delta.c [hunk 1]
```diff
@@ -28,68 +23,46 @@ struct modem_delta_header {
...
 			}
-			if (errno == ENOEXEC) {
-				err = get_modem_error();
-				if (err != DFU_ERASE_PENDING) {
-					LOG_ERR("DFU error: %d", err);
-				}
+			if (err == NRF_MODEM_DELTA_DFU_ERASE_PENDING) {
 				k_sleep(K_SECONDS(SLEEP_TIME));
+			} else {
+				LOG_ERR("Error during erase, error %d", err);
```
[the remark this answers is not in the document]
Comment (anhmolt) [reply subsys/dfu/dfu_target/src/dfu_target_modem_delta.c:66]: Added a `k_sleep(K_SECONDS(SLEEP_TIME))` before the while loop in `delete_banked_modem_delta_fw` function.
Initially giving the modem some more time to react to scratch area erase request seems to have done the trick.

Thanks to @eivindj-nordic for figuring it out!

**Review by eivindj-nordic — APPROVED** [review]

**Comment by lemrey** [comment]
We should update the manifest here, so that it points to https://github.com/lemrey/sdk-zephyr/pull/1.
That way, a `west update` using `libmodem` manifest would import this repository and your commit to `sdk-zephyr` with it.
Once we are ready, we'll merge the `sdk-zephyr` PR, then update the SHA here, then update the SHA in `libmodem`.

**Review by lemrey — COMMENTED** [review]
Looks good, can you make sure this is the same commit as the latest you have in the PR to `sdk-nrf` before we merge?

File: subsys/dfu/dfu_target/src/dfu_target_modem_delta.c [hunk 2]
```diff
@@ -169,104 +101,89 @@ int dfu_target_modem_delta_init(size_t file_size, dfu_target_callback_t cb)
...
-		} else {
-			return 0;
-		}
-	case DFU_AREA_NOT_BLANK:
-		delete_banked_modem_delta_fw();
-		err = dfu_target_modem_delta_write(buf, len);
-		if (err < 0) {
+	err = nrf_modem_delta_dfu_write(buf, len);
+	if (err < 0) {
+		LOG_ERR("Write failed, libmodem error %d", err);
```
Comment (eivindj-nordic) [inline subsys/dfu/dfu_target/src/dfu_target_modem_delta.c:144]: `libmodem` is internal, use `modem library` or` nrf_modem_lib` etc.

**Comment by lemrey** [comment]
Update Zephyr SHA: e37c4108bd3a04c300228f5b17c6d1c8943f1a53

**Pull request merged**